contracts/tgrade-valset/src/contract.rs

```diff
@@ -540,7 +546,27 @@ fn end_block(deps: DepsMut, env: Env) -> Result<Response, ContractError> {
     let old_validators = VALIDATORS.load(deps.storage)?;
     VALIDATORS.save(deps.storage, &validators)?;
     // determine the diff to send back to tendermint
-    let (diff, update_members) = calculate_diff(validators, old_validators);
+    let (diff, add, remove) = calculate_diff(validators, old_validators);
```

Comment:
Alternatively, you could use `diff` below, with the convention that zero power means removal.

That way you can keep the `calculate_diff` signature the same, and keep rewards distribution computation internally.